setup/ocm_addons_test.go

```diff
@@ -0,0 +1,245 @@
+					{
+						Id:    "notification-email",
+						Value: "example@example.com",
+					},
+				}
+
+				err := testutils.SaveAsJsonToArtifactsDir(payload, payloadFileName)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("install RHODS addon", func() {
+				url := fmt.Sprintf("/api/clusters_mgmt/v1/clusters/%s/addons", *ocmClusterId)
+				cmd := fmt.Sprintf("post %s --body=%s", url, payloadPath)
+				out, err := runOcmCommand(cmd)
+				Expect(err).ToNot(HaveOccurred())
+				err = testutils.SaveToArtifactsDir([]byte(out), "ocm-addon-install-resp.json")
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+
+		It("wait for rhods addon to be installed", func() {
+			installed, err := waitForAddonToBeInstalled(*ocmClusterId, rhodsAddonId)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(installed).To(BeTrue(), "addon was not installed correctly")
+		})
+	})
+
+	Context("GPU addon installation", Ordered, func() {
+		var (
+			addonInstalled bool
+		)
+		It("get state of gpu-addon installation", func() {
+			resp := getAddon(*ocmClusterId, gpuAddonId)
+			Expect(resp.Kind).ToNot(BeEmpty())
+			if resp.Kind == "AddOnInstallation" {
+				testutils.Printf("info", "GPU Addon seems to already be installed")
+				addonInstalled = true
+			} else {
+				testutils.Printf("info", "GPU Addon not found in cluster.")
+			}
+			err := testutils.SaveAsJsonToArtifactsDir(resp, "gpu-addon-initial-response.json")
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		Context("install addon if needed", Ordered, func() {
+			var (
+				payloadFileName = "ocm-gpu-addon-payload.json"
+				payloadPath     = fmt.Sprintf("%s/%s", internal.Config.ArtifactDir, payloadFileName)
+			)
+			BeforeAll(func() {
+				if addonInstalled {
+					Skip("GPU addon is already installed on cluster")
+					return
+				}
+			})
+
+			It("prepare addon payload", func() {
+				payload := &ocmAddonPayload{}
+				payload.Addon.Id = gpuAddonId
+
+				err := testutils.SaveAsJsonToArtifactsDir(payload, payloadFileName)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("install gpu addon", func() {
+				url := fmt.Sprintf("/api/clusters_mgmt/v1/clusters/%s/addons", *ocmClusterId)
+				cmd := fmt.Sprintf("post %s --body=%s", url, payloadPath)
+				out, err := runOcmCommand(cmd)
+				Expect(err).ToNot(HaveOccurred())
+				err = testutils.SaveToArtifactsDir([]byte(out), "ocm-addon-install-resp.json")
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+
+		It("wait for gpu addon to be installed", func() {
+			installed, err := waitForAddonToBeInstalled(*ocmClusterId, gpuAddonId)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(installed).To(BeTrue(), "addon was not installed correctly")
+		})
+	})
+
+})
+
+func getAddon(clusterId string, addonId string) *ocmAddonResponse {
+	cmd := fmt.Sprintf("get /api/clusters_mgmt/v1/clusters/%s/addons/%s", clusterId, addonId)
+	out, err := runOcmCommand(cmd)
+	resp := &ocmAddonResponse{}
+	if err != nil {
+		err = json.Unmarshal([]byte(err.Error()), resp)
```

Comment:
Do we care about this `err`? If yes, then we should do something about it, e.g. log  it or return it. Otherwise, we could just use the [blank identifier](https://go.dev/doc/effective_go#blank):

```go
_ = json.Umarshal([]byte(err.Error()), resp)
```